internal/service/thirdpartyintegration/resource_third_party_integration_test.go

```diff
@@ -143,3 +160,212 @@ func testAccCheckMongoDBAtlasThirdPartyIntegrationImportStateIDFunc(resourceName
 		return fmt.Sprintf("%s-%s", ids["project_id"], ids["type"]), nil
 	}
 }
+
+const (
+	Unknown3rdParty = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+	}
+	`
+
+	PAGERDUTY = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		service_key = "%[4]s"
+	}
+	`
+
+	DATADOG = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		api_key = "%[4]s"
+		region  ="%[5]s"
+	}
+	`
+
+	OPSGENIE = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		api_key = "%[4]s"
+		region  = "%[5]s"
+	}
+	`
+	VICTOROPS = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		api_key = "%[4]s"
+		routing_key = "%[5]s"
+	}
+	`
+
+	MICROSOFTTEAMS = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		microsoft_teams_webhook_url = "%[4]s"	
+	}
+	`
+
+	PROMETHEUS = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		user_name = "%[4]s"	
+		password  = "%[5]s"
+		service_discovery = "%[6]s" 
+		scheme = "%[7]s"
+		enabled = true
+	}
+	`
+
+	WEBHOOK = `
+	resource "mongodbatlas_third_party_integration" "%[1]s" {
+		project_id = "%[2]s"
+		type = "%[3]s"
+		url = "%[4]s"	
+	}
+`
+	alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	numeric  = "0123456789"
+	alphaNum = alphabet + numeric
+)
+
+type thirdPartyConfig struct {
+	Name        string
+	ProjectID   string
+	Integration matlas.ThirdPartyIntegration
+}
+
+func configBasic(cfg *thirdPartyConfig) string {
+	dataStr := fmt.Sprintf(`
+		data "mongodbatlas_third_party_integration" %[1]q {
+			project_id = mongodbatlas_third_party_integration.%[1]s.project_id
+			type = mongodbatlas_third_party_integration.%[1]s.type
+		}
+
+		data "mongodbatlas_third_party_integrations" "test" {
+			project_id = mongodbatlas_third_party_integration.%[1]s.project_id
+		}
+	`, cfg.Name)
+
+	switch cfg.Integration.Type {
+	case "PAGER_DUTY":
+		return fmt.Sprintf(PAGERDUTY,
+			cfg.Name,
+			cfg.ProjectID,
+			cfg.Integration.Type,
+			cfg.Integration.ServiceKey,
+		) + dataStr
+	case "DATADOG":
+		return fmt.Sprintf(DATADOG,
+			cfg.Name,
+			cfg.ProjectID,
+			cfg.Integration.Type,
+			cfg.Integration.APIKey,
+			cfg.Integration.Region,
+		) + dataStr
+	case "OPS_GENIE":
```

Comment:
Looks to me that this is the only `cfg.Type` being used?
Seems a bit weird to have all these different types but only test one of them 🤔

Reply:
yeah, i suppose this was tested changing manually in local.  i've prefered to leave that as it was in case we want to improve it later and test all tools